frontend/src/metabase/admin/upsells/components/UpsellCard.tsx

```diff
@@ -18,6 +18,20 @@ type UpsellCardProps = {
   children: React.ReactNode;
 };
 
+interface FullWidthVariant {
+  maxWidth?: never;
+  fullWidth?: boolean;
+}
+
+interface FixWidthVariant {
+  maxWidth?: number;
+  fullWidth?: never;
+}
+
+type Variants = FullWidthVariant | FixWidthVariant;
```

Comment:
This is a very nice way to ensure we don't mistakenly pass both `maxWidth` and `fullWidth` at the same time. It's like an enum, but we also need value from `maxWidth`.

Another potential API could be
```ts
interface Variant {
  width?: number | "fullwidth"
}
```

Let me know what you think.

Reply:
I like what you've done with `never` here ❤️